src/plots/mapbox/layout_attributes.js

```diff
@@ -132,7 +132,7 @@ var attrs = module.exports = overrideAll({
 
         type: {
             valType: 'enumerated',
-            values: ['circle', 'line', 'fill', 'symbol'],
+            values: ['circle', 'line', 'fill', 'symbol', 'raster'],
```

Comment:
As `type: 'raster'` is the value that "makes sense" for `sourcetype: 'raster'`, we should probably override (wrong) user settings during the defaults step:

https://github.com/plotly/plotly.js/blob/c0cebfdd9efaba1cbe3bd550f8569b9d95693793/src/plots/mapbox/layout_defaults.js#L53-L90

Down the road, it might be nice to expose some of mapbox-gl's raster style settings:

https://docs.mapbox.com/mapbox-gl-js/style-spec/#layers-raster